{
  RetryingHttpService service1=createRetryingService(0);
  service1.close();
  RetryingHttpService service2=createRetryingService(1);
  service2.close();
  try (RetryingHttpService service3=createRetryingService(-1)){
    Assert.fail("The argument is not legal ao an exception should be thrown.");
  }
 catch (  IllegalArgumentException exception) {
    Assert.assertTrue(exception.getMessage().contains("-1"));
  }
}
